src/Simplify.elm

```diff
@@ -6163,6 +6182,72 @@ wrapperRepeatChecks wrapper checkInfo =
             Nothing
 
 
+arrayLengthChecks : CheckInfo -> Maybe (Error {})
+arrayLengthChecks checkInfo =
+    firstThatConstructsJust
+        [ \() -> collectionSizeChecks arrayCollection checkInfo
+        , \() -> arrayLengthOnArrayRepeatOrInitializeChecks checkInfo
+        ]
+        ()
+
+
+arrayLengthOnArrayRepeatOrInitializeChecks : CheckInfo -> Maybe (Error {})
+arrayLengthOnArrayRepeatOrInitializeChecks checkInfo =
+    let
+        maybeCall : Maybe ( String, { nodeRange : Range, fnRange : Range, firstArg : Node Expression, argsAfterFirst : List (Node Expression) } )
+        maybeCall =
+            firstThatConstructsJust
+                [ \() ->
+                    AstHelpers.getSpecificFunctionCall ( [ "Array" ], "repeat" ) checkInfo.lookupTable checkInfo.firstArg
+                        |> Maybe.map (Tuple.pair "repeat")
+                , \() ->
+                    AstHelpers.getSpecificFunctionCall ( [ "Array" ], "initialize" ) checkInfo.lookupTable checkInfo.firstArg
+                        |> Maybe.map (Tuple.pair "initialize")
+                ]
+                ()
+    in
+    case maybeCall of
+        Just ( fnName, call ) ->
+            let
+                replacement : { details : List String, fix : List Fix }
+                replacement =
+                    case Evaluate.getInt checkInfo call.firstArg of
+                        Just length ->
+                            let
+                                positiveLength : String
+                                positiveLength =
+                                    String.fromInt (max 0 length)
```

Comment:
But why not do this in `arrayDetermineLength`?

Reply:
Oversight, this helped simplify the code. Keep calling me out, it works great! :smile: (just to clarify, this isn't sarcasm)